databases/week3/mealSharing.sql

```diff
@@ -0,0 +1,450 @@
+DELETE FROM
+  Reservation
+WHERE
+  id = 5;
+INSERT INTO
+  Review (
+    title,
+    `description`,
+    meal_id,
+    stars,
+    created_date
+  )
+VALUES
+  ('nice', 'nice experience', 3, 5, '2021-07-25');
+INSERT INTO
+  Review (
+    title,
+    `description`,
+    meal_id,
+    stars,
+    created_date
+  )
+VALUES
+  (
+    'good',
+    'lovely food with best experience',
+    4,
+    4,
+    '2021-07-26'
+  );
+INSERT INTO
+  Review (
+    title,
+    `description`,
+    meal_id,
+    stars,
+    created_date
+  )
+VALUES
+  ('tastes good', 'awesome.', 2, 4, '2021-07-27');
+INSERT INTO
+  Review (
+    title,
+    `description`,
+    meal_id,
+    stars,
+    created_date
+  )
+VALUES
+  ('fine', 'delicious food', 2, 2, '2021-07-28');
+INSERT INTO
+  Review (
+    title,
+    `description`,
+    meal_id,
+    stars,
+    created_date
+  )
+VALUES
+  (
+    'best',
+    'enjoyed every bit of it',
+    6,
+    3,
+    '2021-07-28'
+  );
+select
+  *
+FROM
+  Review;
+-- Get a Review with any id, fx 1
+SELECT
+  *
+from
+  Review
+WHERE
+  id = 4;
+-- Update a Review with any id, fx 1. Update any attribute fx the title or multiple attributes
+UPDATE
+  Review
+SET
+  `title` = 'super',
+  `stars` = 5
+WHERE
+  id = 3;
+-- Delete a Review with any id, fx 1
+DELETE FROM
+  Review
+WHERE
+  id = 5;
+-- Additional queries
+  -- Get meals that has a price smaller than a specific price fx 90
+SELECT
+  *
+FROM
+  Meal
+WHERE
+  price < 350;
+-- Get meals that still has available reservations
+SELECT
+  Meal.id,
+  Meal.title,
+  Meal.max_reservations,
+  Reservation.id AS reservation_id,
+  Reservation.number_of_guests
+FROM
+  Meal
+  LEFT JOIN Reservation ON Reservation.meal_id = Meal.id
+WHERE
+  Meal.max_reservations > (
+    SELECT
+      SUM(number_of_guests)
+    FROM
+      Reservation
+    WHERE
+      Reservation.meal_id = Meal.id
+  )
+ORDER BY
+  Meal.id;
+-- Get meals that partially match a title. Rød grød med will match the meal with the title Rød grød med fløde
+SELECT
+  *
+FROM
+  Meal
+WHERE
+  title LIKE '%burger%';
+-- Get meals that has been created between two dates
+SELECT
+  *
+FROM
+  Meal
+WHERE
+  created_date > '2021-08-12'
+  AND created_date < '2021-07-15';
+-- Get only specific number of meals fx return only 5 meals
+SELECT
+  *
+FROM
+  Meal
+ORDER BY
+  price
+LIMIT
+  5;
+-- Get the meals that have good reviews
+SELECT
+  *
+FROM
+  Meal
+WHERE
+  id IN (
+    SELECT
+      meal_id
+    FROM
+      Review
+    WHERE
+      stars = 4
+  );
+-- Get reservations for a specific meal sorted by created_date
+SELECT
+  *
+FROM
+  Reservation
+WHERE
+  meal_id IN (
+    SELECT
+      id
+    FROM
+      Meal
+    WHERE
+      title = 'fried chicken'
+  )
+ORDER BY
+  created_date;
+-- Sort all meals by average number of stars in the reviews
+SELECT
+  Meal.id,
+  Meal.title,
+  Review.title,
+  Review.stars
+FROM
+  Meal
+  RIGHT JOIN Review ON Meal.id = Review.meal_id
+ORDER BY
+  Review.stars;
```

Comment:
This should use the Average of the stars from all the reviews of a meal

Reply:
yeah.. fixed now